src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -290,13 +290,33 @@ uct_ib_mlx5_ep_set_rdma_seg(struct mlx5_wqe_raddr_seg *raddr, uint64_t rdma_radd
 }
 
 
+static UCS_F_ALWAYS_INLINE void
+uct_ib_mlx5_set_dgram_seg_grh(struct mlx5_wqe_datagram_seg *seg,
+                              struct mlx5_grh_av *grh_av)
+{
+    struct mlx5_base_av *to_av    = mlx5_av_base(&seg->av);
+    struct mlx5_grh_av *to_grh_av = mlx5_av_grh(&seg->av);
+
+    if (grh_av != NULL) {
+        ucs_assert(to_av->dqp_dct & UCT_IB_MLX5_EXTENDED_UD_AV);
+#if HAVE_STRUCT_MLX5_GRH_AV_RMAC
+        memcpy(to_grh_av->rmac, grh_av->rmac, sizeof(to_grh_av->rmac));
+#endif
+        to_grh_av->tclass     = grh_av->tclass;
+        to_grh_av->hop_limit  = grh_av->hop_limit;
+        to_grh_av->grh_gid_fl = grh_av->grh_gid_fl;
+        memcpy(to_grh_av->rgid, grh_av->rgid, sizeof(to_grh_av->rgid));
+    } else if (to_av->dqp_dct & UCT_IB_MLX5_EXTENDED_UD_AV) {
+        to_grh_av->grh_gid_fl = 0;
+    }
+}
+
 static UCS_F_ALWAYS_INLINE void
 uct_ib_mlx5_set_dgram_seg(struct mlx5_wqe_datagram_seg *seg,
                           uct_ib_mlx5_base_av_t *av, struct mlx5_grh_av *grh_av,
                           int qp_type)
 {
-    struct mlx5_base_av *to_av    = mlx5_av_base(&seg->av);
-    struct mlx5_grh_av *to_grh_av = mlx5_av_grh(&seg->av);
+    struct mlx5_base_av *to_av = mlx5_av_base(&seg->av);
```

Comment:
i think this function (uct_ib_mlx5_set_dgram_seg) is only used by UD now. so can move it to ud C file and clean it up a bit (don't check qp type)

Reply:
> i think this function (uct_ib_mlx5_set_dgram_seg) is only used by UD now. so can move it to ud C file and clean it up a bit (don't check qp type)

ok, done